x-pack/plugins/reporting/server/core.ts

```diff
@@ -121,13 +120,18 @@ export class ReportingCore {
     const config = createConfig(core, context.config.get<ReportingConfigType>(), logger);
     this.config = config;
 
+    this.pdfExport = new PdfExportType(this.core, this.config, this.logger, this.context);
+    this.exportTypesRegistry.register(this.pdfExport);
+
     this.deprecatedAllowedRoles = config.roles.enabled ? config.roles.allow : false;
     this.executeTask = new ExecuteReportTask(this, config, this.logger);
     this.monitorTask = new MonitorReportsTask(this, config, this.logger);
 
     this.getContract = () => ({
       usesUiCapabilities: () => config.roles.enabled === false,
       registerExportTypes: (id) => id,
+      getScreenshots: (options) => this.getScreenshots(options),
+      getSpaceId: (request) => this.getSpaceId(request),
```

Comment:
It seems typescript doesn't like the way this is called. I guess we can't pass `options` without knowing if it is for PNG params or PDF params. I changed these lines a bit to go around the issue:
```suggestion
      getScreenshots: this.getScreenshots.bind(this),
      getSpaceId: this.getSpaceId.bind(this),
```